Makefile

```diff
@@ -1,155 +1,143 @@
-#*******************************************************************************
-#   Ledger App
-#   (c) 2017 Ledger
+# ****************************************************************************
+#    Ledger App Boilerplate
+#    (c) 2020 Ledger SAS.
 #
-#  Licensed under the Apache License, Version 2.0 (the "License");
-#  you may not use this file except in compliance with the License.
-#  You may obtain a copy of the License at
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
 #
-#      http://www.apache.org/licenses/LICENSE-2.0
+#       http://www.apache.org/licenses/LICENSE-2.0
 #
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-#*******************************************************************************
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+# ****************************************************************************
+
+ifeq ($(BOLOS_SDK),)
+    # `THIS_DIR` must be resolved BEFORE any `include` directives
+    THIS_DIR   := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+		TARGET_SDK := $(shell ./util/read-last-sdk)
+		BOLOS_SDK  := ${$(TARGET_SDK)}
+endif
 
 ifeq ($(BOLOS_SDK),)
 $(error Environment variable BOLOS_SDK is not set)
 endif
-include $(BOLOS_SDK)/Makefile.defines
 
-APPVERSION_M=1
-APPVERSION_N=2
-APPVERSION_P=0
-APPVERSION=$(APPVERSION_M).$(APPVERSION_N).$(APPVERSION_P)
-APPNAME = "Solana"
+include $(BOLOS_SDK)/Makefile.defines
 
-APP_LOAD_PARAMS = --curve ed25519 --path "44'/501'" --appFlags 0x240 $(COMMON_LOAD_PARAMS)
```

Comment:
Flag 0x40 will be dropped, no adverse side effect?

Reply:
there shouldn't be any adverse effect.  the app never queries the pin directly.  i couldn't find any justification for why we were requesting this ability in the first place

Reply:
I _did_ fix the incorrect use of assign vs. append when specifying `--appFlags` though :wink: